test/index.js

```diff
@@ -616,8 +616,9 @@ describe('def', function() {
 
     throws(function() { defaultTo(0, 'XXX'); },
            errorEq(TypeError,
-                   '‘defaultTo’ expected a value of type (Nullable Number) ' +
-                   'as its second argument; received "XXX"'));
+                   '‘defaultTo’ expected its first and second arguments to ' +
+                   'be of the same type; the first is of type Number' +
+                   ', however the second, "XXX", is of type String'));
```

Comment:
Let's consider a function such as [`S.fromMaybe :: a -> Maybe a -> a`][1], because I find reasoning about `Nullable a` difficult due to Nullable not being a container.

It's incorrect to state that the first and second arguments must share a type. There is a constraint, but it's not easy to express in words that the value inside the Maybe must share a type with the value provided as the first argument. Perhaps the most succinct approach is to present that type signature and state that the `a`s don't match. We could elaborate by listing the types of each `a` value and stating that the intersection between these sets is empty.

For example:

```
> S.fromMaybe(0, S.Just('XXX'))
TypeError: TypeVariable constraint violation

fromMaybe :: a -> Maybe a -> a
             ^          ^
             1          2

The value at position 1 is 0, which is a member of the following types:

  - Number
  - sanctuary-def/ValidNumber
  - sanctuary-def/Integer

The value at position 2 is "XXX", which is a member of the following
types:

  - String

Since there is no type of which both values are members, the
type-variable constraint has been violated.
```

Now *this* is an error message that would fill me with hope for our profession. :)


[1]: https://github.com/plaid/sanctuary#fromMaybe